From a80cc14bd48e798bb422b4a81e523033834f8d90 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sat, 31 Oct 2026 10:18:28 +0100
Subject: [PATCH 104/N] mimalloc: avoid duplicate process initialization on
 MinGW

CRT/TLS mode already initializes mimalloc through Windows TLS callbacks.
Define MI_PRIM_HAS_PROCESS_ATTACH on MinGW too, preventing the fallback
constructor from initializing it again and flushing a stray newline to
stderr.

Assume native TLS support in the supported MinGW toolchains.
---
 Objects/mimalloc/prim/windows/prim.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Objects/mimalloc/prim/windows/prim.c b/Objects/mimalloc/prim/windows/prim.c
index 5d531de..5e8c607 100644
--- a/Objects/mimalloc/prim/windows/prim.c
+++ b/Objects/mimalloc/prim/windows/prim.c
@@ -775,9 +775,7 @@ static void NTAPI mi_win_main(PVOID module, DWORD reason, LPVOID reserved) {
 #endif
 
 #if defined(MI_WIN_INIT_USE_CRT_TLS)
-  #if !defined(__MINGW32__) || defined(MI_MINGW_UCRT64)  // on mingw without UCRT use the constructor attribute (in `src/prim/prim.c`)
   #define MI_PRIM_HAS_PROCESS_ATTACH  1
-  #endif
 
   // nothing to do since `_mi_thread_done` is handled through the DLL_THREAD_DETACH event.
   void _mi_prim_thread_init_auto_done(void) {}
